htdocs/comm/propal/list.php

```diff
@@ -1270,6 +1310,65 @@
 	$delallowed = $user->rights->propal->creer;
 
 	print $formfile->showdocuments('massfilesarea_proposals', '', $filedir, $urlsource, 0, $delallowed, '', 1, 1, 0, 48, 1, $param, $title, '', '', '', null, $hidegeneratedfilelistifempty);
+
+	if ($action == 'validate') {
+		if (GETPOST('confirm') == 'yes') {
+			foreach ($toselect as $checked) {
+				$sql = "SELECT ref, fk_statut AS status FROM ".MAIN_DB_PREFIX."propal WHERE rowid = ".$checked;
+				$resql = $db->query($sql);
+				if ($resql){
+					$obj = $db->fetch_object($resql);
+					$ref = substr($obj->ref, 1, 4);
+					if ($ref == 'PROV') {
+						$numref = $object->getNextNumRef($soc);
+						if (empty($numref)) {
+							setEventMessages($object->error, $object->errors, 'errors');
+						}
+					} else {
+						$numref = $obj->ref;
+					}
+					if ($obj->status == 0){
+						$sql = "UPDATE ".MAIN_DB_PREFIX."propal SET fk_statut = 1, ref ='".$numref."' WHERE rowid = ".$checked;
```

Comment:
Using a direct update is not a good choice. Because it means trigger won't be executed, actions of external modules won't work.
The select ref, fk_status must be replace by a 
$tmpproposal->fetch($checked)
Then we must call 
$tmpproposal->validate($user);